src/main/kotlin/no/nav/klage/oppgave/eventlisteners/StatistikkTilDVHEventListener.kt

```diff
@@ -0,0 +1,112 @@
+
+@Service
+class StatistikkTilDVHEventListener(
+    private val statistikkTilDVHKafkaProducer: StatistikkTilDVHKafkaProducer,
+    private val klagebehandlingRepository: KlagebehandlingRepository,
+    private val mottakRepository: MottakRepository
+) {
+
+    companion object {
+        @Suppress("JAVA_CLASS_ON_COMPANION")
+        private val logger = getLogger(javaClass.enclosingClass)
+    }
+
+    @EventListener
+    fun klagebehandlingEndretEventToDVH(klagebehandlingEndretEvent: KlagebehandlingEndretEvent) {
+        logger.debug("Received KlagebehandlingEndretEvent for klagebehandlingId ${klagebehandlingEndretEvent.klagebehandling.id} in StatistikkTilDVHEventListener")
+
+        val klagebehandling =
+            klagebehandlingRepository.findById(klagebehandlingEndretEvent.klagebehandling.id).orElseThrow()
+
+        val mottak = mottakRepository.getOne(klagebehandling.mottakId)
+
+        if (shouldSendStats(klagebehandlingEndretEvent.endringslogginnslag)) {
+            val klageStatistikkTilDVH = createDTO(
+                klagebehandling,
+                mottak,
+                getKlagebehandlingState(klagebehandlingEndretEvent.endringslogginnslag)
+            )
+            statistikkTilDVHKafkaProducer.sendStatistikkTilDVH(klageStatistikkTilDVH)
+        }
+    }
+
+    private fun shouldSendStats(endringslogginnslag: List<Endringslogginnslag>) =
+        endringslogginnslag.isEmpty() ||
+                endringslogginnslag.any { it.felt === Felt.TILDELT_SAKSBEHANDLERIDENT || it.felt === Felt.SLUTTFOERT }
+
+    private fun getKlagebehandlingState(endringslogginnslag: List<Endringslogginnslag>): String {
+        return when {
+            endringslogginnslag.isEmpty() -> "MOTTATT"
+            endringslogginnslag.any { it.felt === Felt.TILDELT_SAKSBEHANDLERIDENT } -> "TILDELT_SAKSBEHANDLER"
+            endringslogginnslag.any { it.felt === Felt.SLUTTFOERT } -> "SLUTTFOERT"
+            else -> "UNKNOWN".also {
+                logger.warn(
+                    "unknown state for klagebehandling with id {}",
+                    endringslogginnslag.first().klagebehandlingId
+                )
+            }
+        }
+    }
+
+    private fun createDTO(
+        klagebehandling: Klagebehandling,
+        mottak: Mottak,
+        klagebehandlingState: String
+    ): KlageStatistikkTilDVH {
+        //Only works as long as we only have one
+        val vedtak = klagebehandling.vedtak.firstOrNull()
+        val now = LocalDateTime.now()
+
+        return KlageStatistikkTilDVH(
+            ansvarligEnhetKode = klagebehandling.tildeltEnhet,
+            behandlingId = mottak.dvhReferanse,
+            behandlingIdKabal = klagebehandling.id.toString(),
+            behandlingStartetKA = klagebehandling.startet,
+            behandlingStatus = klagebehandlingState,
+            behandlingType = klagebehandling.type.navn,
+            beslutter = klagebehandling.medunderskriverident,
+            endringstid = now,
```

Comment:
Dette er den funksjonelle tiden? Blir det riktig å sette den til now? Burde den ikke vært hentet fra domenet? (Jeg er generelt ikke så glad i now.. ;) )

Reply:
`modified` blir det riktige å bruke her ja.

Reply:
Du svarer det samme på begge spørsmålene om tidspunkt, og da blir jeg litt usikker. Hvis vi bruker modified i begge tilfeller, så klarer vi jo ikke å skille på funksjonell og teknisk tid?

Reply:
Stemmer at vi ikke skiller på de nå. Men:

"Ved for eksempel patching av data eller oppdatering tilbake i tid, skal tekniskTid være lik endrings tidspunktet, mens funksjonellTid (som vi nå kaller `endringstid`..) angir tidspunktet da endringen offisielt gjelder fra."

Mulig vi burde brukt `modified` til tekniskTid, og `"finalized" e.l.` til endringstid (funksjonellTid).